### PR TITLE
test(prs-556): add integration tests and docs for deploy-time posture

### PR DESCRIPTION
**Pull request by pepe-anchor**
> Replaces #444, which GitHub closed as merged when it was merged into its parent
> stack branch instead of being rebased.

## Why am I making this PR?

The deploy-time posture needs end-to-end integration test evidence and the published docs must reflect the new requirement.

## What am I changing?

Four new integration tests covering all posture+signature combinations. Fix two vacuous tests that were passing under the wrong posture. Update docs: library getting-started, add-a-chain guide, dApp integration page, and gRPC server page all document the deploy-time posture choice.

One change from the original #444: the `examples/library_integration_test.rs`
call-site update for `create_registry(&ParserConfig)` now lands in the parser_app
commit, which is where that signature changes. #444 left the example uncompilable
across three commits.

## What is the Linear ticket?

PRS-556

## What are the rollback steps?

Revert the commit.

## Is this change backwards compatible?

Yes, tests and docs only.

## Does this require cross-team/service coordination?

No.

## How do I know it works as designed? Which tests exercise this code?

The four new integration tests. Existing doc build pipeline verifies snippets compile.

Verified at this commit: `cargo clippy --all-targets -- -D warnings` clean, and the
tree here is byte-identical to the pre-rebuild stack tip.

### CODE REVIEW

File: src/integration/tests/parser.rs [hunk 1]
```diff
@@ -910,6 +913,190 @@ async fn parser_near_metadata_network_reaches_the_account_suffix_check_e2e() {
...
+
+/// A parser deployed with `--accept-unsigned-abis` honours the unsigned ABI: the
+/// calldata is decoded and the function name shows up in the payload.
+#[tokio::test]
+async fn accept_unsigned_abis_deployment_decodes_unsigned_abi() {
+    async fn test(test_args: TestArgs) {
+        let payload = parsed_payload_for_unsigned_abi(test_args).await;
+        assert!(
+            payload.contains("frobnicate"),
+            "accept-unsigned deployment should decode the unsigned ABI, got: {payload}"
```
[the remark this answers is not in the document]
Comment (pepe-anchor) [reply src/integration/tests/parser.rs:1123]: Added `accept_unsigned_abis_deployment_decodes_foreign_signed_abi`: sends the FOREIGN_SIGNER_SIG/PUBKEY mapping to a plain `integration::Builder::new()` (permissive) deployment and asserts it decodes, the counterpart to `require_signed_abis_deployment_drops_foreign_signed_abi`.

File: docs/wallet-integration/grpc-server/getting-started.mdx [hunk 2]
```diff
@@ -50,6 +50,36 @@ EPHEMERAL_FILE=/path/to/key.secret cargo run --bin parser_grpc_server
 
 For development, the binary defaults to `integration/fixtures/ephemeral.secret`.
 
+#### ABI trust posture
+
+Whether caller-supplied Ethereum `abi_mappings` are honoured when they carry no
+signature is chosen at startup, not per request. Pass exactly one of:
```
[the remark this answers is not in the document]
Comment (pepe-anchor) [reply docs/wallet-integration/grpc-server/getting-started.mdx]: Fair catch, reworded to a recommendation ("We recommend passing exactly one of these flags explicitly, rather than relying on the default described below") so it no longer reads as a requirement that the Warning immediately contradicts.

**Comment by pepe-anchor** [comment]
Addressed Copilot's review summary (review 5265666800):\n\n| Item | Verdict | What happened |\n|---|---|---|\n| (none) | skip | no actionable items beyond the two inline threads, both already fixed and resolved |\n\n

File: src/integration/tests/parser.rs [hunk 3]
```diff
@@ -910,6 +913,213 @@ async fn parser_near_metadata_network_reaches_the_account_suffix_check_e2e() {
...
 }
 
+// ---------------------------------------------------------------------------
+// Deploy-time ABI trust posture (PRS-556)
+// ---------------------------------------------------------------------------
+//
+// The two tests below send the SAME request to two parser_app instances that
+// differ only in the cmdline they were started with. That is the property the
+// deploy-time flag buys: whether an unverified ABI is honoured is decided by the
+// deployment, not by what the caller put in (or left out of) the request.
```
[the remark this answers is not in the document]
Comment (pepe-anchor) [reply src/integration/tests/parser.rs]: Fixed: the header undercounted (it said two tests when there are actually five: 2 permissive + 3 strict). Rewrote it to name the posture pairs and the allowlisted-signer acceptance case that has no permissive counterpart, instead of restating a count.

**Comment by pepe-anchor** [comment]
Addressed Copilot's review summary (review 5265783840):

| Item | Verdict | What happened |
|------|---------|----------------|
| Update test header to describe four posture tests | fix | Same finding as the inline thread on line 923; fixed there (header actually undercounted at 5 tests, not 4/2, rewrote it to name the posture pairs). |

**Comment by pepe-anchor** [comment]
Addressed Copilot's review summary (review 5265875793):

| Item | Verdict | What happened |
|------|---------|----------------|
| Scope `.gitignore` so `.pi/extensions/` stays trackable | fix | Removed the redundant blanket `.pi/` line this branch added (`.gitignore` line 29); the existing `.pi/mega-compact/` entry already covers per-user pi scratch, and the blanket line was shadowing new untracked files under `.pi/extensions/`, contradicting the comment above it. |

**Comment by pepe-anchor** [comment]
Addressed Copilot's review summary (review 5266071990):

| Item | Verdict | What happened |
|------|---------|----------------|
| Remove or relocate the unused `MetadataTrustPolicy` import in the library getting-started example | fix | Removed the dead `use visualsign::signing::MetadataTrustPolicy;` at line 44 of `docs/wallet-integration/library/getting-started.mdx` ("Basic usage" example, which uses `ParserConfig::accept_unsigned()` and never references the type). The second, separate import further down the file (used by the "ABI trust posture" example) is untouched, it's still needed there. |

**Comment by pepe-anchor** [comment]
Addressed Copilot's review summary (review 5266300948):

| Item | Verdict | What happened |
|---|---|---|
| Add invalid-signature integration coverage | fix | The 4 existing posture tests only covered unsigned and validly-signed-but-wrong-signer cases. Added `accept_unsigned_abis_deployment_drops_invalidly_signed_abi` and `require_signed_abis_deployment_drops_invalidly_signed_abi` in `src/integration/tests/parser.rs`, using a signature with one flipped byte (well-formed DER, fails verification), asserting the ABI entry is dropped under both deploy-time postures. |
| Align signing instructions with parser's domain-separated prehash | fix | Confirmed against `src/visualsign/src/signing.rs` (`METADATA_SIGNING_DOMAIN_V1 = "visualsign-metadata-v1"`, scope = 8-byte big-endian chain ID + 20-byte contract address) that the doc's Node.js example only hashed the raw ABI JSON. Rewrote the example in `docs/wallet-integration/dapps/ethereum.mdx` to build and sign the actual length-prefixed, domain-separated preimage, and updated the "Hash" bullet to describe it. Verified the new JS example produces a digest byte-identical to the parser's own `ethereum_metadata_prehash` for the same inputs. |

File: src/integration/tests/parser.rs [hunk 4]
```diff
@@ -910,6 +913,243 @@ async fn parser_near_metadata_network_reaches_the_account_suffix_check_e2e() {
...
+// ---------------------------------------------------------------------------
+// Deploy-time ABI trust posture (PRS-556)
+// ---------------------------------------------------------------------------
+//
+// The tests below send the SAME requests to parser_app instances that differ
+// only in the cmdline they were started with. That is the property the
+// deploy-time flag buys: whether an unverified ABI is honoured is decided by the
+// deployment, not by what the caller put in (or left out of) the request.
+//
+// Two posture pairs, plus one posture-specific acceptance case:
```
[the remark this answers is not in the document]
Comment (pepe-anchor) [reply src/integration/tests/parser.rs]: Fixed: changed the heading to "Three posture pairs" at line 925, it does document unsigned, foreign-signed, and invalidly-signed pairs plus the allowlisted-signed acceptance case. Confirmed via grep there's only one occurrence of this heading in the file (the line 950 reference in the review was stale).

**Comment by pepe-anchor** [comment]
Addressed Copilot's review summary (review 5266622181):

| Item | Verdict | What happened |
|------|---------|---------------|
| Ethereum signing example uses an invalid placeholder address (docs/wallet-integration/dapps/ethereum.mdx:98) | fix | Replaced `"0xYourContractAddress"` with a syntactically valid 40-hex-char placeholder (`0x1234...7890`) plus a comment to substitute the real contract address; the old string wasn't hex and would silently truncate through `Buffer.from(...,"hex")`. |

**Comment by pepe-anchor** [comment]
Addressed Copilot's review summary (review 5266702746):

| Item | Verdict | What happened |
|------|---------|----------------|
| (none) | skip | no actionable items found in the review body |

**Review by shahan-khatchadourian-anchorage — APPROVED** [review]
Reviewed with a full multi-agent pass (per-file, deep-logic, security, context/history). No blocking issues — approving.

**Strengths**
1. Posture test matrix is complete: all four bypass regressions (flag ignored, allowlist dropped, verification skipped, everything rejected) are each caught by a test, plus an acceptance case so "reject everything" can't pass as "enforce allowlist."
2. Checked-in signature fixtures are real — independently recomputed the domain-separated prehash and verified each DER signature against its pubkey.
3. `ethereum.mdx`'s prehash writeup is byte-exact and reimplementable, and honestly documents the silent-drop failure mode instead of glossing over it.

**Non-blocking findings**
1. `ethereum.mdx:147,149` overstates two security properties: says the posture "cannot be detected" from the dApp side (actually `tvc-deploy`'s `pivotArgs` manifest exposes it, per #455), and says a present signature under accept-unsigned "proves the ABI was not modified" (it only catches a naive mutation, not a full entry rewrite). Two stale source comments (`abi_metadata.rs:66-68`, `cli_plugin.rs:52-54`) plus the root `CLAUDE.md` all still say this wiring "hasn't landed yet" — likely the source of the doc error, fix all three together.
2. JS signing example's `contractAddress.slice(2)` assumes a `0x` prefix; without it you get a silently-wrong 19-byte address and a signature that never verifies, with no diagnostic. Also tells integrators to sign "the contract address" rather than whatever key the wallet registers the entry under (wrong for proxies).
3. PR description says "four new tests" / "fix two vacuous tests" — diff actually adds seven tests and modifies none. Worth correcting so reviewers don't credit coverage that isn't here.
4. Integration harness gaps: `require_signed_abis` replaces rather than appends `abi_trust_args` (multi-signer allowlist untested end-to-end); no test starts `parser_app` with zero/both posture flags (core "refuses to start" guarantee only unit-tested); `execute()` doesn't check the child before the 90s port-wait timeout (startup-rejection regressions surface as an opaque panic); four new helper functions are near-duplicates and could collapse to one.
5. `docs/wallet-integration/chains/ethereum.mdx` (untouched by this PR) still describes the signature as optional — worth a follow-up since wallets populate `abi_mappings`.
